File: src/openfl/events/TextEvent.nim
import ./Event

type
  TextEvent* = ref object of Event
    text*: string

proc newTextEvent*(typeStr: string, bubbles: bool = false, cancelable: bool = false, text: string = ""): TextEvent =
  let self = TextEvent(text: text)
  self.`type` = typeStr
  self.bubbles = bubbles
  self.cancelable = cancelable
  return self

method clone*(self: TextEvent): Event =
  let event = newTextEvent(self.`type`, self.bubbles, self.cancelable, self.text)
  event.eventPhase = self.eventPhase
  event.target = self.target
  event.currentTarget = self.currentTarget
  return event

method toString*(self: TextEvent): string =
  "[TextEvent type=" & self.`type` & " bubbles=" & $self.bubbles & " cancelable=" & $self.cancelable & " text=" & self.text & "]"

const
  LINK* = "link"
  TEXT_INPUT* = "textInput"

File: kfl/src/main/kotlin/openfl/events/TextEvent.kt
package openfl.events

open class TextEvent(
    type: String,
    bubbles: Boolean = false,
    cancelable: Boolean = false,
    var text: String = ""
) : Event(type, bubbles, cancelable) {

    companion object {
        const val LINK = "link"
        const val TEXT_INPUT = "textInput"
    }

    override fun clone(): TextEvent {
        val event = TextEvent(type, bubbles, cancelable, text)
        event.target = target
        event.currentTarget = currentTarget
        event.eventPhase = eventPhase
        return event
    }

    override fun toString(): String {
        return formatToString("TextEvent", "type", "bubbles", "cancelable", "text")
    }
}
